tensorflow_model_optimization/g3doc/guide/clustering/index.md

```diff
@@ -0,0 +1,99 @@
+# Weight clustering
+
+This document provides an overview on weight clustering to help you determine how it fits with your use case. To dive right into the code, see the [weight clustering end-to-end example](clustering_example.ipynb) and the [API docs](../../api_docs/python). For additional details on how to use the Keras API, a deep dive into weight clustering, and documentation on more advanced usage patterns, see the [weight clustering comprehensive guide](clustering_comprehensive_guide.ipynb).
+
+## Overview
+
+Clustering, or weight sharing, reduces the number of unique weight values in a model, leading to benefits for deployment. It first groups the weights of each layer into *N* clusters, then shares the cluster's centroid value for all the weights belonging to the cluster.
+
+This technique brings improvements in terms of model compression. By reducing the number of unique weight values, weight clustering renders the weights suitable for compression via Huffman coding and similar techniques. Future framework support will, therefore, be able to reduce memory size and bandwidth requirements. This can be critical for deploying deep learning models on embedded systems with limited resources.
```

Comment:
1. Given comment below on "We have seen up to 5x improvements" which will make things much shorter, I think you can fold that content back into this paragraph up here. 

2. "By reducing the number of unique weight values, weight clustering renders the weights suitable for compression via Huffman coding and similar techniques."
I don't think this is needed. The intuition is stated in the first paragraph already. Huffman coding is a part of the "how" and doesn't need to be mentioned in the doc. 
 
E.g. 

This technique brings improvements via model compression. Future framework support can unlock memory footprint improvements. We've seen up to 5x improvements in model compression with minimal loss of accuracy.